subsys/net/pkt_filter/CMakeLists.txt

```diff
@@ -7,3 +7,9 @@ zephyr_library_sources(base.c)
 zephyr_library_sources_ifdef(CONFIG_NET_L2_ETHERNET ethernet.c)
 
 endif()
+
+if(CONFIG_NET_FILTER)
+zephyr_library()
```

Comment:
Please double check - I'm not actually sure if two libraries can be defined in a single file (especially if you don't provide the library name).

Reply:
I move it to another directory with separate cmake file and Kconfig.